### PR TITLE
boards: arm: Add STM32F3DISCOVERY board

### PR DESCRIPTION
**Pull request by ydamigos**
Add support for STM32F3DISCOVERY board.

The following samples were tested:
    hello_world
    basic/button
    basic/blinky

### CODE REVIEW

**Review by galak — CHANGES_REQUESTED** [review]
Missing board yaml file

**Review by galak — CHANGES_REQUESTED** [review]
Merge the dts patch with the board patch

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+   $<zephyr_root_path>/samples/hello_world
+
+To build the Zephyr kernel and application, enter:
+
+.. code-block:: console
+
+   $ cd <zephyr_root_path>
+   $ source zephyr-env.sh
+   $ cd $ZEPHYR_BASE/samples/hello_world/
+   $ make BOARD=stm32f4_disco
```
Comment (dbkinder) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: should that be `stm32f3_disco`  (3 vs. 4)?

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+Connect the STM32F3DISCOVERY Discovery kit to your host computer using the USB
+ST-LINK port. Then, enter the following command:
+
+.. code-block:: console
+
+   $ make BOARD=stm32f4_disco flash
```
Comment (dbkinder) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: should that be `stm32f3_disco`  (3 vs. 4)?

**Review by dbkinder — CHANGES_REQUESTED** [review]
a couple of typos (I think)

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst (hunk 1 above)
Comment (ydamigos) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: it should be stm32f3_disco

File: boards/arm/stm32f3_disco/doc/stm32f3_disco.rst (hunk 2 above)
Comment (ydamigos) [inline boards/arm/stm32f3_disco/doc/stm32f3_disco.rst]: as above

**Comment by ydamigos** [comment]
@galak the yaml file was already in the board patch. I squashed the dts patch in the board patch.

**Comment by galak** [comment]
@dbkinder docs look good?

File: boards/arm/stm32f3_disco/stm32f3_disco_defconfig [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# ARM Options
+CONFIG_CPU_HAS_FPU=y
+
+# Floating Point Options
+CONFIG_FLOAT=y
+
+# enable uart driver
+CONFIG_SERIAL=y
+CONFIG_UART_STM32=y
+# enable USART2 - passthrough to STLINK v2-1 connector
```
Comment (erwango) [inline boards/arm/stm32f3_disco/stm32f3_disco_defconfig]: Typo: USART1

**Review by erwango — CHANGES_REQUESTED** [review]
One really minor comment.
Otherwise LGTM

File: boards/arm/stm32f3_disco/stm32f3_disco_defconfig (hunk 3 above)
Comment (ydamigos) [inline boards/arm/stm32f3_disco/stm32f3_disco_defconfig]: Fixed

**Comment by galak** [comment]
@dbkinder docs good now?

**Review by dbkinder — APPROVED** [review]
+1

**Review by galak — APPROVED** [review]

**Pull request merged**